app/models/concerns/omniauthable.rb

```
@@ -0,0 +1,37 @@
+        # Update provider and uid for existing user (first-time linking)
+        user.update!(provider: provider, uid: uid)
+        return user
+      end
+
+      # Create new user if not found
+      user = create(
+        email: data['email'],
+        password: Devise.friendly_token[0, 20],
+        provider: provider,
+        uid: uid
+      )
```

RuboCop (1.81.7)

[convention] Redundant assignment before returning detected.

(Style/RedundantAssignment)